pkg/nix/nix.go

```diff
@@ -38,20 +38,28 @@ func IsInstalled() bool {
 func BuildDependencies(deps []Dependency, cache *Cache) (_ DependenciesToStorePathMap, err error) {
 	defer errz.Recover(&err)
 
+	var unSatisfiedDeps []Dependency
 	pkgToStorePath := make(DependenciesToStorePathMap)
 
 	for _, v := range deps {
-		var key string
-
 		if cache != nil {
-			key, err = GenerateKey(v)
+			key, err := GenerateKey(v)
 			errz.Fatal(err)
+
 			if storePath, ok := cache.Get(key); ok {
 				pkgToStorePath[v] = StorePath(storePath)
 				continue
 			}
+			unSatisfiedDeps = append(unSatisfiedDeps, v)
 		}
+	}
+
+	if len(unSatisfiedDeps) > 0 {
+		fmt.Println("Building nix dependencies...")
```

Comment:
Should we use a boblog.Log call here instead of fmt?

Reply:
As discussed with @Equanox , logging should be handled like this:

- bob.Log() should be used for logging
- fmt.PrintX should be used for user output
- printX() should be used for debugging
Closing.